### PR TITLE
Add FXIOS-3456 [v97] Adjust with SKAdNetwork

### PR DESCRIPTION
**Pull request by lmarceau**
# Overview 
Adding Adjust SDK for ticket [FXIOS-3456](https://mozilla-hub.atlassian.net/browse/FXIOS-3456)

### CODE REVIEW

File: Client/AdjustHelper.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        Adjust.appDidLaunch(adjustConfig)
+    }
+
+    private static func getBundle() -> Bundle {
+        var bundle = Bundle.main
+        if bundle.bundleURL.pathExtension == "appex" {
+            // Peel off two directory levels - MY_APP.app/PlugIns/MY_APP_EXTENSION.appex
+            let url = bundle.bundleURL.deletingLastPathComponent().deletingLastPathComponent()
+            if let extensionBundle = Bundle(url: url) {
+                bundle = extensionBundle
```
Comment (nbhasin2) [inline Client/AdjustHelper.swift:35]: This is fine however is it possible to use what we have in AppInfo

https://github.com/mozilla-mobile/firefox-ios/blob/main/Shared/AppInfo.swift#L16-L20

**Review by nbhasin2 — APPROVED** [review]

File: Client/AdjustHelper.swift (hunk 1 above)
Comment (lmarceau) [reply Client/AdjustHelper.swift]: Done :) Also adapted Sentry class so it's consistent

**Review by nbhasin2 — APPROVED** [review]

**Comment by lmarceau** [comment]
Just waiting some feedback on functionalities for this feature then we should be able to move this forward.

File: Client.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -7531,7 +7548,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "/usr/local/bin/carthage copy-frameworks";
+			shellScript = "/usr/local/bin/carthage copy-frameworks\n";
```
Comment (nbhasin2) [inline Client.xcodeproj/project.pbxproj:7551]:
```suggestion
			shellScript = "/usr/local/bin/carthage copy-frameworks";
```

File: Client.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -7545,7 +7562,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if [ \"${INCLUDE_SETTINGS_BUNDLE}\" = \"YES\" ]\nthen\n    cp -r \"${PROJECT_DIR}/${TARGET_NAME}/Application/Settings.bundle\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\"\nfi";
+			shellScript = "if [ \"${INCLUDE_SETTINGS_BUNDLE}\" = \"YES\" ]\nthen\n    cp -r \"${PROJECT_DIR}/${TARGET_NAME}/Application/Settings.bundle\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\"\nfi\n";
```
Comment (nbhasin2) [inline Client.xcodeproj/project.pbxproj:7565]:
```suggestion
			shellScript = "if [ \"${INCLUDE_SETTINGS_BUNDLE}\" = \"YES\" ]\nthen\n    cp -r \"${PROJECT_DIR}/${TARGET_NAME}/Application/Settings.bundle\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\"\nfi";
```

File: Client.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -7561,7 +7578,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "/usr/local/bin/carthage copy-frameworks";
+			shellScript = "/usr/local/bin/carthage copy-frameworks\n";
```
Comment (nbhasin2) [inline Client.xcodeproj/project.pbxproj:7581]:
```suggestion
			shellScript = "/usr/local/bin/carthage copy-frameworks";
```

File: Client.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 5]
```diff
@@ -2,6 +2,6 @@
 <Workspace
    version = "1.0">
    <FileRef
-      location = "self:Client.xcodeproj">
+      location = "self:">
```
Comment (nbhasin2) [inline Client.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]:
```suggestion
      location = "self:Client.xcodeproj">
```
This file should technically be added to gitignore but since its already added lets revert the changes

File: Client/AdjustHelper.swift [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        // initialized through appDidFinishLaunching otherwise it will be in a bad state.
+        Adjust.appDidLaunch(config)
+
+        AdjustHelper.setEnabled(shouldEnable)
+    }
+
+    /// Used to enable or disable Adjust SDK and it's features. We disable third party sharing by default.
+    /// If user has disabled Send Anonymous Usage Data then we ask Adjust to erase the user's data as well.
+    static func setEnabled(_ enabled: Bool) {
+        Adjust.disableThirdPartySharing()
```
Comment (nbhasin2) [inline Client/AdjustHelper.swift:33]: Q. Considering we are always disabling first. Have we asked or looked at their api that it's okay to disable it first?

ref: https://github.com/adjust/android_sdk/issues/408 - This is an older issue but its good to document what we do here.

File: package-lock.json [hunk 7]
```diff
@@ -5,6 +5,7 @@
   "requires": true,
   "packages": {
     "": {
+      "name": "firefox-ios",
```
Comment (nbhasin2) [inline package-lock.json:8]: nit: this doesn't really add any value and from what I recall it gets generated with npm 🤔

File: Client/AdjustHelper.swift (hunk 6 above)
Comment (lmarceau) [reply Client/AdjustHelper.swift:33]: Yes, it was determined it was ok for us to disable by default third party sharing after receiving the following answer from Adjust:

> The disableThirdPartySharing functionality is typically used for apps that give users the option to opt out of Third 
> Party Sharing, and is usually implemented on the user level. If this is implemented for all users, it limits the ability 
> to share attribution data with partners (such as Facebook and Google) and will limit their ability to receive 
> callbacks from Adjust, directly affecting the attribution ability of partners.

Since this use case doesn't apply to us, it was decided to disable by default.

**Review by nbhasin2 — APPROVED** [review]

**Comment by lmarceau** [comment]
Build passed but didn't update here: https://app.bitrise.io/build/81c8bd71-1a58-49b9-a6fb-815cd88f17b8#?tab=log
Screen Shot 2022-01-24 at 4 27 19 PM
